libconcurrent/primitives/barrier.c

```diff
@@ -1,16 +1,45 @@
 #include <barrier.h>
 
-inline void BarrierInit(Barrier *bar, int n) {
-    bar->v = n;
+inline void BarrierSet(Barrier *bar, uint32_t n) {
```

Comment:
We can discuss the design of a barrier that can be set once and used multiple times, just like the POSIX API.

Reply:
Actually, the last implementation of our barrier, can be set once and used multiple times (i.e., re-entrant barrier).

Reply:
Correct. Good job.